runtime/compiler/optimizer/DataAccessAccelerator.cpp

```diff
@@ -95,7 +95,10 @@ int32_t TR_DataAccessAccelerator::perform()
        !comp()->getOption(TR_MimicInterpreterFrameShape) &&
 
        // We cannot handle arraylets because hardware intrinsics act on contiguous memory
-       !comp()->generateArraylets()&& !TR::Compiler->om.useHybridArraylets())
+       !comp()->generateArraylets()&& !TR::Compiler->om.useHybridArraylets() &&
+       // AOT can not relocate direct calls in the DAA OOL paths. If AOT is on and DAA makes
+       // a call in the OOL path, the target won't be reached and it seg-faults. Disable DAA for now.
+       !(comp()->compileRelocatableCode() && !comp()->getOption(TR_UseSymbolValidationManager)))
```

Comment:
@fjeremic  We won't need to revert this back for SVM as this optimization will be disabled when in AOT it does not use SVM.  When we looked into the failure, it seems like AOT is doing right thing (Relocationg correct resolved ramCP) but fails in VM routine resolving this DAA method which we still don't understand why and needs some more digging.

Reply:
@NigelYiboYu can you please dump your investigation details in an issue so we can tag someone from the VM side to help with the investigation if you are seeing crashes in the VM resolve code?

Reply:
I'd also like to the comment here changed to reference this issue you will open. I'd like to see this fixed eventually and this code reverted.

Reply:
VM helper crash issue is here: https://github.com/eclipse/openj9/issues/4363

Reply:
Thanks @NigelYiboYu can you please `TODO (#4363): AOT can not relocate...` directly in the code so that we know which issue to look at in the future to get this fixed?

Reply:
Added the TODO tag.